src/nvim/os/env.c

```diff
@@ -187,45 +187,63 @@ void init_homedir(void)
 
   const char *var = os_getenv("HOME");
 
-#ifdef WIN32
-  // Typically, $HOME is not defined on Windows, unless the user has
-  // specifically defined it for Vim's sake. However, on Windows NT
-  // platforms, $HOMEDRIVE and $HOMEPATH are automatically defined for
-  // each user. Try constructing $HOME from these.
-  if (var == NULL) {
-    const char *homedrive = os_getenv("HOMEDRIVE");
-    const char *homepath = os_getenv("HOMEPATH");
-    if (homepath == NULL) {
-      homepath = "\\";
-    }
-    if (homedrive != NULL
-        && strlen(homedrive) + strlen(homepath) < MAXPATHL) {
-      snprintf(os_buf, MAXPATHL, "%s%s", homedrive, homepath);
-      if (os_buf[0] != NUL) {
-        var = os_buf;
+#ifdef UNIX
+  // use uv_os_homedir only for unix systems
+  // as uv_os_homedir checks %USERPROFILE% first then only %HOMEDRIVE%%HOMEPATH%  
```

Comment:
My initial commit was to use `uv_os_homedir()` for both Windows and Unix systems and fallback to the old procedure in case it fails. But as @janlazo pointed out, 
>Whatever happens here, the new-style oldtests from patches 8.0.1012 and 8.0.1017 should pass with no changes unless you modify the environment variable priority (ie. `%USERPROFILE%` is checked first before `%HOMEDRIVE%%HOMEPATH%%`).

and
 
> #8614 doesn't affect Windows because it doesn't rely on `HOME` to begin with. Vim depends on it for `~/vimfiles` so it returns the value of the static variable, `homedir`, when HOME is empty/unset. Neovim shares the same behaviour for compatibility. It's convenient for me since I use the same config for both via junctions/symlinks.

so `uv_os_homedir()` is only used for Unix systems, with a fallback to the old code in case it fails. Although, I don't have an error log added in case it does fail.